lib/os/fdtable.c

```diff
@@ -304,10 +304,13 @@ ssize_t zvfs_read(int fd, void *buf, size_t sz)
 		return -1;
 	}
 
-	(void)k_mutex_lock(&fdtable[fd].lock, K_FOREVER);
+	if (fdtable[fd].vtable->read == NULL) {
```

Comment:
Does this check belong inside the lock? From this vantage point, it looks like it should be.

Reply:
Good catch.

I was thinking the `vtable->read` is `const` for a given vtable, but of course it is `vtable` itself is not const

Reply:
Fixed!

Reply:
looks good!